rasa/core/channels/mattermost.py

```diff
@@ -70,13 +136,77 @@ def __init__(self, url: Text, team: Text, user: Text, pw: Text) -> None:
             team: Your mattermost team name
             user: Your mattermost userid that will post messages
             pw: Your mattermost password for your user
+            webhook_url: The mattermost callback url as specified
+                in the outgoing webhooks in mattermost example
+                https://mysite.example.com/webhooks/mattermost/webhook
         """
         self.url = url
         self.team = team
         self.user = user
         self.pw = pw
+        self.webhook_url = webhook_url
 
-    def blueprint(self, on_new_message):
+    async def message_with_trigger_word(
+        self,
+        on_new_message: Callable[[UserMessage], Awaitable[None]],
+        output: Dict[Text, Any],
+    ) -> None:
+        # splitting to get rid of the @botmention
+        # trigger we are using for this
+        text = output["text"].split(" ", 1)
+        text = text[1]
+
+        sender_id = output["user_id"]
+        self.bot_channel = output["channel_id"]
+
+        try:
+            out_channel = MattermostBot(
+                self.url,
+                self.team,
+                self.user,
+                self.pw,
+                self.bot_channel,
+                self.webhook_url,
+            )
+            user_msg = UserMessage(
+                text, out_channel, sender_id, input_channel=self.name()
+            )
+            await on_new_message(user_msg)
+        except Exception as e:
+            logger.error("Exception when trying to handle message.{0}".format(e))
+            logger.debug(e, exc_info=True)
+            pass
```

Comment:
Can be removed.

Reply:
Are you sure? Where will the exception be handled if I remove this?

Reply:
I meant, you can remove the `pass`, not the exception :)

Reply:
Alright. That's true. Removed it in both places.